server/main.py

```diff
@@ -96,67 +102,132 @@ def __init__(self) -> None:
-        self._rooms[room_name]["clients"].add(client)
+        Args:
+            client: The Client to which the connection belongs.
+            room_code: The room from which the client will be disconnected.
+        """
+        if self.room_exists(room_code):
+            self._rooms[room_code]["clients"].add(client)
+        else:
+            raise RoomNotFoundError(f"The room with code '{room_code}' was not found.", KappaCloseCodes.RoomNotFound)
 
-    def disconnect(self, client: Client, room_name: str) -> None:
+    def disconnect(self, client: Client, room_code: str) -> None:
         """Removes the connection from the active connections.
 
         If, after the disconnection, the room is empty, delete it.
 
         Args:
             client: The Client to which the connection belongs.
-            room: The room from which the client will be disconnected.
+            room_code: The room from which the client will be disconnected.
         """
-        self._rooms[room_name]["clients"].remove(client)
+        self._rooms[room_code]["clients"].remove(client)
 
-        if len(self._rooms[room_name]["clients"]) == 0:
-            del self._rooms[room_name]
+        if len(self._rooms[room_code]["clients"]) == 0:
+            del self._rooms[room_code]
 
-    async def broadcast(self, data: dict, room_name: str, sender: Client | None = None) -> None:
+    async def broadcast(self, data: dict, room_code: str, sender: Client | None = None) -> None:
         """Broadcasts data to all active connections.
 
         Args:
             data: The data to be sent to the clients, it should always contain a
                 "type" key to indicate the event type.
-            room: The room to which the data will be sent.
+            room_code: The room to which the data will be sent.
             sender (optional): The client who sent the message.
         """
-        for connection in self._rooms[room_name]["clients"]:
+        for connection in self._rooms[room_code]["clients"]:
             if connection == sender:
                 continue
             await connection.send(data)
 
+    def room_exists(self, room_code: str) -> bool:
+        """Checks if a room exists.
+
+        Args:
+            room_code: The code associated with a particular room.
+
+        Returns:
+            True if the room exists. False otherwise.
+        """
+        if room_code in self._rooms:
+            return True
+        return False
+
 
 manager = ConnectionManager()
 
 
-@app.websocket("/room/{room_name}")
-async def room(websocket: WebSocket, room_name: str) -> None:
+class ConnectionEventData(BaseModel):
+    """A model representing the connection event data."""
+
+    message: str
+    difficulty: int
+    room_code: str
+    connection_type: Literal["create", "join"]
+
+
+class ConnectionData(BaseModel):
+    """A model representing the initial connection."""
+
+    type: str
+    data: ConnectionEventData
+
+
+@app.websocket("/room/")
```

Comment:
I would suggest removing the slash after `room`, right now typing it in the URL is necessary.